File: UI/src/services/warehouseService.js
// Warehouse service — maps GET /api/Warehouse
// API WarehouseItemDto fields match UI shape exactly:
//   id (uuid string), name, qty (double), unit, location, category, note (all string|null)
import { get } from './api.js';

const normalise = (w) => ({
  ...w,
  name:     w.name     ?? '',
  unit:     w.unit     ?? '',
  location: w.location ?? '',
  category: w.category ?? '',
  note:     w.note     ?? '',
});

export const getAll = () =>
  get('/Warehouse').then((data) => data.map(normalise));

File: API/PalletTimeLine.Api/Application/Services/WarehouseService.cs
using Microsoft.EntityFrameworkCore;
using PalletTimeLine.Api.Application.DTOs;
using PalletTimeLine.Api.Infrastructure.Data;

namespace PalletTimeLine.Api.Application.Services;

public class WarehouseService : IWarehouseService
{
    private readonly PalletTimelineDbContext _db;

    public WarehouseService(PalletTimelineDbContext db)
    {
        _db = db;
    }

    public async Task<IReadOnlyList<WarehouseItemDto>> GetAllAsync(CancellationToken ct = default)
    {
        return await _db.WarehouseItems.AsNoTracking()
            .Select(i => new WarehouseItemDto(i.Id, i.Name, i.Qty, i.Unit, i.Location, i.Category, i.Note))
            .ToListAsync(ct);
    }
}
